test/logintest.js

```diff
@@ -0,0 +1,143 @@
+const request = require('supertest');
+const app = require('../api/server').app;
+const assert = require('assert');
+const db = require('../api/models');
+const config = require('../api/config');
+const clone = require('clone');
+const fs = require('fs');
+
+const errorMessage = "Error: This is a top-10 common password - Add another word or two. Uncommon words are better.. Please choose a stronger password and try again.";
+describe('Signup API', () => {
+  before((done) => {
+    // Connect to test database
+    db.init(() => {
+      // Clear test database
+      dbConnection = db.init(config.mongodb_test, done);
+      db.mongo.User.deleteMany({}, (err) => {
+        if (err) return done(err);
+        done();
+      });
+    });
+  });
+
+  after((done) => {
+    // Disconnect from test database
+    db.disconnect(done());
+  });
+
+  afterEach((done) => {
+    //  Clear test database after test case
+    db.mongo.User.deleteMany({}, (err) => {
+      if (err) return done(err);
+    });
+    done();
+  })
+
+  it('should return error when username already exists', (done) => {
+    // Create a test user with a username test username
+    let testUser = {
+      username: 'testuser',
+      email: 'testuser@example.com',
+      password: 'password!@123',
+    };
+    db.mongo.User.create(testUser, (err, user) => {
+      if (err) return done(err);
+      // Make a POST request to signup API with the same username
+      request(app)
+        .post('/signup')
+        .send({
+          username: 'testuser',
+          email: 'newuser@example.com',
+          password: 'Testpassword',
+        })
+        .expect(500)
+        .end((err, res) => {
+          if (err) return done(err);
+          assert.equal(res.body.message,
+            'The username you chose is already registered. If it is yours, please try signing in, or register with a different username.');
+          done();
+        });
+    });
+  });
+
+  it('should return error for common password', (done) => {
+    // Make a POST request to signup API with a new username and email
+    request(app)
+      .post('/signup')
+      .send({
+        username: 'newuser',
+        email: 'newuser@example.com',
+        password: 'password',
+      })
+      .expect(500)
+      .end((err, res) => {
+        if (err) return done(err);
+        assert.equal(res.body.message, errorMessage)
+        done();
+      });
```

Comment:
does `request` support an async/await interface? to avoid the callback in the `.end`

something like
```
const res = request(app).post....
```